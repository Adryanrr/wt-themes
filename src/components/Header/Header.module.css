.title {
  position: sticky;
  top: 0;
  z-index: 1;
  text-align: center;
  font-size: 24px;
  margin: 0;
}

.paragraph {
  padding: 0 10px;
}

.download {
  color: #7fd8ff;
  text-decoration: underline;
  cursor: pointer;
  background: transparent;
  border: 0;
  font-size: inherit;
  padding: 0;
}
