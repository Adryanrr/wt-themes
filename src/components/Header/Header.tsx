import React from 'react';

import css from './Header.module.css';

const Header: React.FC = () => (
  <>
    <h1 className={css.title}>Windows Terminal Themes</h1>
    <p className={css.paragraph}>
      Themes for{' '}
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://www.microsoft.com/en-us/p/windows-terminal-preview/9n0dx20hk701"
      >
        Windows Terminal (Preview)
      </a>
      . To add new themes, open up settings (profile.json), copy a theme into
      schemes and then reference the name in profiles.
    </p>
    <p className={css.paragraph}>
      The themes come from{' '}
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://github.com/mbadolato/iTerm2-Color-Schemes"
      >
        iTerm2 Color Schemes
      </a>
      , so thanks to them.
    </p>
    <p className={css.paragraph}>
      <a
        target="_blank"
        rel="noopener noreferrer"
        href={`${process.env.REACT_APP_PUBLIC_PATH}/colour-schemes.json`}
      >
        Download all the themes
      </a>
      {' | '}
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://github.com/atomcorp/themes"
      >
        Github page
      </a>
    </p>
  </>
);

export default Header;
