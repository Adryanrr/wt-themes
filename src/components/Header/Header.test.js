import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from './Header';

it('renders component', () => {
  const {getByText, getByRole} = render(<Header />);
  // console.log(getByText('Windows Terminal Themes'), getByRole('heading'));
  expect(getByRole('heading')).toHaveTextContent('Windows Terminal Themes');
  expect(getByText('Windows Terminal (Preview)')).toMatchSnapshot();
  expect(getByText('iTerm2 Color Schemes')).toMatchSnapshot();
  expect(getByText('Download all the themes')).toMatchSnapshot();
  expect(getByText('Github page')).toMatchSnapshot();
  // expect(asFragment()).toMatchSnapshot()
});
