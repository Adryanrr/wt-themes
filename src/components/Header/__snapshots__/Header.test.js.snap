// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<a
  href="https://www.microsoft.com/en-us/p/windows-terminal-preview/9n0dx20hk701"
  rel="noopener noreferrer"
  target="_blank"
>
  Windows Terminal (Preview)
</a>
`;

exports[`renders component 2`] = `
<a
  href="https://github.com/mbadolato/iTerm2-Color-Schemes"
  rel="noopener noreferrer"
  target="_blank"
>
  iTerm2 Color Schemes
</a>
`;

exports[`renders component 3`] = `
<a
  href="undefined/colour-schemes.json"
  rel="noopener noreferrer"
  target="_blank"
>
  Download all the themes
</a>
`;

exports[`renders component 4`] = `
<a
  href="https://github.com/atomcorp/themes"
  rel="noopener noreferrer"
  target="_blank"
>
  Github page
</a>
`;
